### PR TITLE
Fixes #1950 - Show a loading indicator while searching for people to …

### PR DESCRIPTION
**Pull request by stefanceriu**
…invite

https://github.com/vector-im/element-x-ios/assets/637564/c72daec0-e0a8-4f05-b2a0-89e607c43fbb

### CODE REVIEW

File: ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenModels.swift [hunk 1]
```diff
@@ -41,12 +41,10 @@ struct InviteUsersScreenViewState: BindableState {
...
     var membershipState: [String: MembershipState] = .init()
     
-    var isSearching: Bool {
-        !bindings.searchQuery.isEmpty
-    }
+    var isSearching = false
     
     var hasEmptySearchResults: Bool {
-        isSearching && usersSection.type == .searchResult && usersSection.users.isEmpty
+        !bindings.searchQuery.isEmpty && usersSection.type == .searchResult && usersSection.users.isEmpty
```
Comment (alfogrillo) [inline ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenModels.swift:47]:
```suggestion
        !isSearching && !bindings.searchQuery.isEmpty && usersSection.type == .searchResult && usersSection.users.isEmpty
```
Shouldn't it be like this?

File: ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenViewModel.swift [hunk 2]
```diff
@@ -134,7 +134,13 @@ class InviteUsersScreenViewModel: InviteUsersScreenViewModelType, InviteUsersScr
             fetchSuggestions()
             return
         }
+        
+        state.isSearching = true
         fetchUsersTask = Task {
+            defer {
+                state.isSearching = false
+            }
```
Comment (alfogrillo) [inline ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenViewModel.swift:142]: Not sure this takes in account Task cancellation.

Assume:
- Task 1 starts
- Task 2 (cancels Task 1)
- Task 1 is canceled but it still run the `defer` messing up the `isSeaching` property (before Task 2 ends)

wdyt?

File: ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenModels.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenModels.swift:47]: Well I didn't want to change the original logic so I copied over what the old `isSearching` used to do ¯\_(ツ)_/¯

Comment (alfogrillo) [reply ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenModels.swift:47]: Got it! :D
I think we use it to show the "empty screen view". So maybe it's better to show it after the search ends?

Comment (stefanceriu) [reply ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenModels.swift:47]: Okay cool, I'll give it a shot then

File: ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenViewModel.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/InviteUsersScreen/InviteUsersScreenViewModel.swift]: Talked about offline, moved the state reset into `handleResult`

**Review by alfogrillo — APPROVED** [review]
LGTM ⏱️

**Pull request merged**